data/src/main/java/org/apache/iceberg/data/DataUtil.java

```diff
@@ -0,0 +1,201 @@
+     * @param uri partition location URI
+     * @param format partition format, avro, parquet or orc
+     * @param spec a partition spec
+     * @param conf a Hadoop conf
+     * @param metricsConfig a metrics conf
+     * @return a List of DataFile
+     */
+  public static List<DataFile> listPartition(Map<String, String> partitionKeys, String uri, String format,
+                                             PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig) {
+    return listPartition(partitionKeys, uri, format, spec, conf, metricsConfig, null);
+  }
+
+    /**
+     * Returns the data files in a partition by listing the partition location.
+     * <p>
+     * For Parquet and ORC partitions, this will read metrics from the file footer. For Avro partitions,
+     * metrics are set to null.
+     * <p>
+     * Note: certain metrics, like NaN counts, that are only supported by iceberg file writers but not file footers,
+     * will not be populated.
+     *
+     * @param partitionKeys partition key, e.g., "a=1/b=2"
+     * @param uri partition location URI
+     * @param format partition format, avro, parquet or orc
+     * @param spec a partition spec
+     * @param conf a Hadoop conf
+     * @param metricsConfig a metrics conf
+     * @param mapping a name mapping
+     * @return a List of DataFile
+     */
+  public static List<DataFile> listPartition(Map<String, String> partitionKeys, String uri, String format,
+                                             PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig,
+                                             NameMapping mapping) {
+    if (format.contains("avro")) {
+      return listAvroPartition(partitionKeys, uri, spec, conf);
+    } else if (format.contains("parquet")) {
+      return listParquetPartition(partitionKeys, uri, spec, conf, metricsConfig, mapping);
+    } else if (format.contains("orc")) {
+      return listOrcPartition(partitionKeys, uri, spec, conf, metricsConfig, mapping);
+    } else {
+      throw new UnsupportedOperationException("Unknown partition format: " + format);
+    }
+  }
+
+  private static List<DataFile> listAvroPartition(Map<String, String> partitionPath, String partitionUri,
+                                                  PartitionSpec spec, Configuration conf) {
+    try {
+      Path partition = new Path(partitionUri);
+      FileSystem fs = partition.getFileSystem(conf);
+      return Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
+              .filter(FileStatus::isFile)
+              .map(stat -> {
+                  // Avro file statistics cannot be calculated without reading the file.
+                  // Setting the rowCount to 0 is just a workaround so that the DataFiles.Builder.build() doesn't fail.
+                Metrics metrics = new Metrics(0L, null, null, null);
+                String partitionKey = spec.fields().stream()
+                          .map(PartitionField::name)
```

Comment:
Iceberg normally uses just 4 spaces for continued indentation. Can we format this class accordingly?